soccer/src/soccer/strategy/agent/position/runner.cpp

```diff
@@ -0,0 +1,80 @@
+#include "runner.hpp"
+
+namespace strategy {
+
+Runner::Runner(int r_id) : Position(r_id) {
+    position_name_ = "Runner";
+    current_state_ = SIDE1;
+}
+
+std::optional<RobotIntent> Runner::derived_get_task(RobotIntent intent) {
+    current_state_ = update_state();
+    return state_to_task(intent);
+}
+
+Runner::State Runner::update_state() {
+    State next_state = current_state_;
+    // handle transitions between current state
+    WorldState* world_state = this->world_state();
+
+    rj_geometry::Point robot_position = world_state->get_robot(true, robot_id_).pose.position();
+  
+     if (current_state_ == SIDE1) {
+      rj_geometry::Point end_position = rj_geometry::Point{3.0, 3.0};
+      double distance_to_end = robot_position.dist_to(end_position);
+      if (distance_to_end < 0.1) {
```

Comment:
thresholds are arbitrary. prefer using check_is_done() to see if a previous motion command has completed. if you actually need different behavior, leave a comment explaining why.